https://gitlab.nic.cz/knot/knot-resolver/-/merge_requests/1697/
From 00929c0b0af776b4a5e344b6f3a0541fefe93db2 Mon Sep 17 00:00:00 2001
From: Brad Cowie <brad@faucet.nz>
Date: Fri, 30 May 2025 13:57:25 +1200
Subject: [PATCH 1/2] datamodel/templates: fix kr_rule_local_* macros

commit a782e9c3 broke the jinja2 generation of the
kr_rule_local_* macro functions. C.KR_RULE_OPTS_DEFAULT
was provided as an argument to the assert() function call
instead of the call to the corresponding C.kr_rule_local_* function
---
 .../templates/macros/local_data_macros.lua.j2          | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/python/knot_resolver/datamodel/templates/macros/local_data_macros.lua.j2 b/python/knot_resolver/datamodel/templates/macros/local_data_macros.lua.j2
index e91b82685..39029b3c7 100644
--- a/python/knot_resolver/datamodel/templates/macros/local_data_macros.lua.j2
+++ b/python/knot_resolver/datamodel/templates/macros/local_data_macros.lua.j2
@@ -28,8 +28,8 @@ hints.root_file('{{ file }}')
 
 {% macro kr_rule_local_address(name, address, nodata, ttl, tags=none) -%}
 assert(C.kr_rule_local_address('{{ name }}', '{{ address }}',
-    {{ boolean(nodata) }}, {{ local_data_ttl(ttl)}}, {{ policy_get_tagset(tags) }}) == 0,
-    C.KR_RULE_OPTS_DEFAULT)
+    {{ boolean(nodata) }}, {{ local_data_ttl(ttl)}}, {{ policy_get_tagset(tags) }},
+    C.KR_RULE_OPTS_DEFAULT) == 0)
 {%- endmacro -%}
 
 
@@ -44,7 +44,7 @@ assert(C.kr_rule_local_address('{{ name }}', '{{ address }}',
 
 {% macro kr_rule_local_hosts(file, nodata, ttl, tags=none) -%}
 assert(C.kr_rule_local_hosts('{{ file }}', {{ boolean(nodata) }},
-    {{ local_data_ttl(ttl)}}, {{ policy_get_tagset(tags) }}) == 0, C.KR_RULE_OPTS_DEFAULT)
+    {{ local_data_ttl(ttl)}}, {{ policy_get_tagset(tags) }}, C.KR_RULE_OPTS_DEFAULT) == 0)
 {%- endmacro %}
 
 
@@ -92,8 +92,8 @@ assert(C.kr_rule_zonefile({{ id }})==0)
 
 {% macro kr_rule_local_subtree(name, type, ttl, tags=none) -%}
 assert(C.kr_rule_local_subtree(todname('{{ name }}'),
-    C.KR_RULE_SUB_{{ type.upper() }}, {{ local_data_ttl(ttl) }}, {{ policy_get_tagset(tags) }}) == 0,
-    C.KR_RULE_OPTS_DEFAULT)
+    C.KR_RULE_SUB_{{ type.upper() }}, {{ local_data_ttl(ttl) }}, {{ policy_get_tagset(tags) }},
+    C.KR_RULE_OPTS_DEFAULT) == 0)
 {%- endmacro %}
 
 
-- 
GitLab

